src/shogun/multiclass/ShareBoost.cpp

```diff
@@ -53,8 +53,8 @@ bool CShareBoost::train_machine(CFeatures* data)
 
 	if (m_nonzero_feas <= 0)
 		SG_ERROR("Set a valid (> 0) number of non-zero features to seek before training\n")
-	if (m_nonzero_feas >= dynamic_cast<CDenseFeatures<float64_t>*>(m_features)->get_num_features())
-		SG_ERROR("It doesn't make sense to use ShareBoost with num non-zero features >= num features in the data\n")
+	if (m_nonzero_feas > dynamic_cast<CDenseFeatures<float64_t>*>(m_features)->get_num_features())
```

Comment:
Please put this into a new PR
Also, if you are touching this already, please do
´"Number of non-zero features (%d) cannot be larger than number of features (%d) in the data\n"`
I.e. pass the information to user

Furthermore, you need to check that the result of the dynamic cast is not NULL, in which case you would segfault here